ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Check if admin ack is required before upgrade
+function admin_ack() {
+    if [[ "${source_minor_version}" -eq "${target_minor_version}" ]]; then
+        echo "Upgrade between z-stream version does not require admin ack" && return
+    fi   
+    
+    local out; out="$(oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data")"
+    if [[ ${out} != *"ack-4.${source_minor_version}"* ]]; then
+        echo "Admin ack not required" && return
+    fi        
+    
+    echo "Require admin ack"
+    local wait_time_loop_var=0 ack_data 
+    ack_data="$(echo $out | awk '{print $2}' | cut -f2 -d\")" && echo "Admin ack patch data is: ${ack_data}"
+    oc -n openshift-config patch configmap admin-acks --patch '{"data":{"'"${ack_data}"'": "true"}}' --type=merge
+    
+    echo "Admin-acks patch gets started"
+            
+    echo -e "sleep 5 min wait admin-acks patch to be valid...\n"
+    while (( wait_time_loop_var < 5 )); do
+        sleep 1m
+        echo -e "wait_time_passed=${wait_time_loop_var} min.\n"
+        if ! oc adm upgrade | grep "AdminAckRequired"; then
+            echo -e "Admin-acks patch PASSED\n"
+            return 0              
+        else
+            echo -e "Admin-acks patch still in processing, waiting...\n"
+        fi
+        (( wait_time_loop_var += 1 ))
+    done
+    if (( wait_time_loop_var >= 5 )); then
+        echo >&2 "Timed out waiting for admin-acks completing, exiting" && return 1
+    fi
+}
+
+# Upgrade the cluster to target release
+function upgrade() {
+    local err current_image
+    
+    if ! err="$(oc adm upgrade --to-image="${target}" 2>&1 >/dev/null)" && [[ $err == *"you must pass --allow-explicit-upgrade to continue"* ]]; then
+        echo "Error msg check on oc adm upgrade --to-image missing --allow-explicit-upgrade PASSED"
+    else
+        echo >&2 "Error msg check on oc adm upgrade --to-image missing --allow-explicit-upgrade FAILED, err: ${err}" && return 1
+    fi
+    
+    current_image="$(oc get clusterversion version -o json|jq -r '.status.history[0].image')"
+    if [[ ${target} == "${current_image}" ]]; then
+        echo >&2 "Upgrade with oc adm upgrade --to-image missing --allow-explicit-upgrade starts but expected NOT start, exiting" && return 1
+    fi
+    
+    if [[ "${FORCE_UPDATE}" == "false" ]]; then
+        oc adm upgrade --to-image="${target}" --allow-explicit-upgrade
```

Comment:
> About oc's version, is it handled by the framework?
Yes.

Since each upgrade has a step, we leverage the `cli` keyword provided by the framework. You can check it in -ref.yaml.

Reply:
`cli: latest`, ack